redbpf/src/lib.rs

```diff
@@ -392,6 +414,61 @@ impl Program {
     }
 }
 
+impl Drop for ProgramData {
+    fn drop(&mut self) {
+        if self.fd.is_some() {
+            unsafe {
+                let _ = libc::close(self.fd.unwrap());
+            }
+        }
+    }
+}
+
+fn pin_bpf_obj<P: AsRef<Path>>(fd: RawFd, pathname: P) -> Result<()> {
+    let parent = pathname.as_ref().parent();
+    if parent.is_none() {
```

Comment:
This happens when `pathname` is `/`? Might be clearer to use `file` as argument name, and potentially a better error than `ErrorKind::InvalidInput`, which is not super descriptive. It might also be worth validating if this is on a bpf file system path based on /proc/mounts for instance.

Reply:
Yes. If `pathname` is `/`, `ErrorKind::InvalidInput` is returned. I will think of other error value. Any suggestions?

In the first patch, if `pathname` is not valid bpf fs, error is returned from `bpf_sys::bpf_obj_pin`
`IO(Os { code: 1, kind: PermissionDenied, message: "Operation not permitted" })`
But it just gives small information to users. And it is difficult to understand root cause.

Validation will help users correct error.